addons/mass_mailing/models/mass_mailing.py

```diff
@@ -196,167 +212,49 @@ def _get_statistics(self, cr, uid, ids, name, arg, context=None):
-            type='integer', multi='_get_statistics'
-        ),
-        'failed': fields.function(
-            _get_statistics, string='Failed',
-            type='integer', multi='_get_statistics'
-        ),
-        'sent': fields.function(
-            _get_statistics, string='Sent Emails',
-            type='integer', multi='_get_statistics'
-        ),
-        'delivered': fields.function(
-            _get_statistics, string='Delivered',
-            type='integer', multi='_get_statistics',
-        ),
-        'opened': fields.function(
-            _get_statistics, string='Opened',
-            type='integer', multi='_get_statistics',
-        ),
-        'replied': fields.function(
-            _get_statistics, string='Replied',
-            type='integer', multi='_get_statistics'
-        ),
-        'bounced': fields.function(
-            _get_statistics, string='Bounced',
-            type='integer', multi='_get_statistics'
-        ),
-        'received_ratio': fields.function(
-            _get_statistics, string='Received Ratio',
-            type='integer', multi='_get_statistics',
-        ),
-        'opened_ratio': fields.function(
-            _get_statistics, string='Opened Ratio',
-            type='integer', multi='_get_statistics',
-        ),
-        'replied_ratio': fields.function(
-            _get_statistics, string='Replied Ratio',
-            type='integer', multi='_get_statistics',
-        ),
-        'total_mailings': fields.function(_get_total_mailings, string='Mailings', type='integer'),
-        'bounced_ratio': fields.function(
-            _get_statistics, string='Bounced Ratio',
-            type='integer', multi='_get_statistics',
-        ),
-    }
-
-    def _get_default_stage_id(self, cr, uid, context=None):
-        stage_ids = self.pool['mail.mass_mailing.stage'].search(cr, uid, [], limit=1, context=context)
-        return stage_ids and stage_ids[0] or False
-
-    def _get_source_id(self, cr, uid, context=None):
-        return self.pool['ir.model.data'].xmlid_to_res_id(cr, uid, 'utm.utm_source_newsletter')
-
-    def _get_medium_id(self, cr, uid, context=None):
-        return self.pool['ir.model.data'].xmlid_to_res_id(cr, uid, 'utm.utm_medium_email')
-
-    _defaults = {
-        'user_id': lambda self, cr, uid, ctx=None: uid,
-        'stage_id': lambda self, *args: self._get_default_stage_id(*args),
-        'source_id': lambda self, *args: self._get_source_id(*args),
-        'medium_id': lambda self,*args: self._get_medium_id(*args),
-    }
-
-    def mass_mailing_statistics_action(self, cr, uid, ids, context=None):
-        res = self.pool['ir.actions.act_window'].for_xml_id(cr, uid, 'mass_mailing', 'action_view_mass_mailing_statistics', context=context)
-        res['domain'] = [('mass_mailing_campaign_id', 'in', ids)]
+    @api.multi
+    def mass_mailing_statistics_action(self):
+        res = self.env.ref('mass_mailing.action_view_mass_mailing_statistics')
```

Comment:
better to use self.env['ir.actions.act_window'].for_xml_id('mass_mailing', 'action_view_mass_mailing_statistics')